### PR TITLE
✨ Added `pyfiction` documentation to read the docs

### PR DESCRIPTION
**Pull request by simon1hofmann**
## Description

Added code tabs to switch between fiction/pyfiction in the online documentation.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [ ] I have added appropriate tests and documentation.
- [x] I have made sure that all CI jobs on GitHub pass.
- [ ] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

**Comment by simon1hofmann** [comment]
@marcelwa Could you please review this PR ?

File: docs/algorithms/apply_gate_library.rst [hunk 1]
```diff
@@ -9,4 +9,10 @@ Applies an FCN :ref:`gate library <fcn-gate-libraries>` to a ``gate_level_layout
 in the same :ref:`technology <fcn-cell-technologies>` as the provided gate library. Thereby, this function creates cell-accurate
 implementations for each gate present in the passed ``gate_level_layout``.
 
-.. doxygenfunction:: fiction::apply_gate_library(const GateLyt& lyt)
+.. tabs::
+    .. tab:: C++
```
Comment (marcelwa) [inline docs/algorithms/apply_gate_library.rst:13]: Would it be possible to use `.. code-tab:: c++` and `.. code-tab:: py` everywhere to have the tabs sync and render code automatically? Or would that break the visuals somewhere?

File: bindings/pyfiction/include/pyfiction/documentation.hpp [hunk 2]
```diff
@@ -3,4 +3,28 @@
 
 #include "pyfiction/pybind11_mkdoc_docstrings.hpp"
 
+static const char *__doc_fiction_cartesian_layout_overridden =
```
Comment (marcelwa) [inline bindings/pyfiction/include/pyfiction/documentation.hpp:6]: Could you do the same for the hexagonal layouts? We use the `even_row` orientation in `pyfiction`. Many thanks!

File: docs/algorithms/orthogonal.rst [hunk 3]
```diff
@@ -9,6 +9,13 @@ Utilizes approximations to the graph-theoretical problem of *Orthogonal Graph Dr
 layouts from logic network specifications. This approach is scalable but requires that the input network is restricted
 to a 3-graph. At the same time, the output layout will always be 2DDWave-clocked and has a large area overhead.
 
-.. doxygenstruct:: fiction::orthogonal_physical_design_params
-   :members:
-.. doxygenfunction:: fiction::orthogonal(const Ntk& ntk, orthogonal_physical_design_params ps = {}, orthogonal_physical_design_stats* pst = nullptr)
+.. tabs::
+    .. tab:: C++
+        .. doxygenstruct:: fiction::orthogonal_physical_design_params
```
Comment (marcelwa) [inline docs/algorithms/orthogonal.rst:14]: Could we pull the header inside the C++ tab here, too, please?

File: docs/layouts/cartesian_layout.rst [hunk 4]
```diff
@@ -5,5 +5,11 @@ Cartesian Layout
 
 Coordinate system that represents a Cartesian grid of compile-time coordinate types.
 
-.. doxygenclass:: fiction::cartesian_layout
-   :members:
+.. tabs::
+    .. tab:: C++
+        .. doxygenclass:: fiction::cartesian_layout
```
Comment (marcelwa) [inline docs/layouts/cartesian_layout.rst:10]: Here too. Can the header go inside the tab?

File: docs/layouts/cell_level_layout.rst [hunk 5]
```diff
@@ -15,5 +15,16 @@ knowledge of their implemented functionality.
 A cell-level layout can be obtained from a ``gate_level_layout`` by the :ref:`application <apply-gate-library>` of a
 :ref:`gate library <fcn-gate-libraries>`.
 
-.. doxygenclass:: fiction::cell_level_layout
-   :members:
+.. tabs::
+    .. tab:: C++
+        .. doxygenclass:: fiction::cell_level_layout
```
Comment (marcelwa) [inline docs/layouts/cell_level_layout.rst:20]: Same here

File: docs/layouts/clocked_layout.rst [hunk 6]
```diff
@@ -13,5 +13,13 @@ The clocked layout can be layered on top of any coordinate (in this context call
 time to extend its functionality by a notion of FCN clocking. Various :ref:`pre-defined clocking schemes<clocking schemes>`
 can be utilized or the clock numbers can be assigned manually.
 
-.. doxygenclass:: fiction::clocked_layout
-   :members:
+.. tabs::
+    .. tab:: C++
+        .. doxygenclass:: fiction::clocked_layout
+           :members:
```
Comment (marcelwa) [inline docs/layouts/clocked_layout.rst:19]: Same here

File: docs/layouts/gate_level_layout.rst [hunk 7]
```diff
@@ -14,6 +14,13 @@ The gate-level layout is implemented in accordance with ``mockturtle``\ 's
 network that can be passed to various of mockturtle's algorithms. However, since each logic node (gate) in the layout
 has to have a concrete position assigned, mockturtle cannot be used to generate valid layouts.
 
-.. doxygenclass:: fiction::gate_level_layout
-   :members:
+.. tabs::
+    .. tab:: C++
+        .. doxygenclass:: fiction::gate_level_layout
```
Comment (marcelwa) [inline docs/layouts/gate_level_layout.rst:19]: same here

File: docs/layouts/obstruction_layout.rst [hunk 8]
```diff
@@ -8,9 +8,17 @@ can be layered on top of any coordinate layout. An obstruction could be due to a
...
 
-.. doxygenclass:: fiction::obstruction_layout
-   :members:
-.. doxygenclass:: fiction::obstruction_layout< Lyt, true >
-   :members:
-.. doxygenclass:: fiction::obstruction_layout< Lyt, false >
-   :members:
+.. tabs::
+    .. tab:: C++
+        .. doxygenclass:: fiction::obstruction_layout
```
Comment (marcelwa) [inline docs/layouts/obstruction_layout.rst:13]: same here

File: docs/requirements.txt [hunk 9]
```diff
@@ -1,2 +1,4 @@
 sphinx>=5.0.0
 breathe>=4.30.0
+sphinx-tabs>=3.4.1
+z3-solver>=4.12.1
```
Comment (marcelwa) [inline docs/requirements.txt:4]: I believe we don't require such a high version of Z3 :thinking: I could be wrong, but `>=4.8` should work fine.

**Review by marcelwa — CHANGES_REQUESTED** [review]

File: docs/algorithms/apply_gate_library.rst (hunk 1 above)
Comment (simon1hofmann) [reply docs/algorithms/apply_gate_library.rst:11]: Unfortunately, `doxygenfunction` and `auto-doc` don't work with code-tabs.

Comment (marcelwa) [reply docs/algorithms/apply_gate_library.rst:11]: No problem. Many thanks for trying!

**Review by marcelwa — APPROVED** [review]
Many thanks!

**Pull request merged**